mRides-app/MapActivity.cs

```diff
@@ -430,20 +468,65 @@ public void OnError(Statuses status)
 
         public void OnPlaceSelected(IPlace place)
         {
-            string destination = place.NameFormatted.ToString();
-            string usr_destination = GetString(Resource.String.dest);
-            Toast.MakeText(ApplicationContext, usr_destination + " : " + destination, ToastLength.Long).Show();         
-        
-            if (destinationMarker != null)
-                map.Clear();
-            destinationMarker = map.AddMarker(new MarkerOptions().SetPosition(place.LatLng).SetTitle(destination));
+            if (!mapButtonClicked)
+            {
+                string destination = place.NameFormatted.ToString();
+                string usr_destination = GetString(Resource.String.dest);
+                string pathURL;
+
+                Toast.MakeText(ApplicationContext, usr_destination + " : " + destination, ToastLength.Long).Show();
+
+                if (destinationMarker != null)
+                {
+                    destinationMarker.Position = place.LatLng;
+                    destinationMarker.Title = destination;
+                }                  
+                else
+                    destinationMarker = map.AddMarker(new MarkerOptions().SetPosition(place.LatLng).SetTitle(destination));
+                if (originMarker != null)
```

Comment:
The brackets aren't consistent at times, sometimes you add them for multiple lines and sometimes there are no brackets (vs line 190) Just making sure it is readable and that there won't be any compile/run-time problems